src/main/scala/tech/cryptonomic/conseil/metadata/MetadataService.scala

```diff
@@ -41,12 +45,23 @@ class MetadataService(config: PlatformsConfiguration,
   }
 
   // fetches table attributes
-  def getTableAttributes(path: EntityPath)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[Attribute]]] = for {
-    exists <- exists(path)
-    attributes <- tezosPlatformDiscoveryOperations.getTableAttributes(path.entity)
-  } yield attributes
-    .filter { _ => exists }
-    .mapNested(attribute => transformation.overrideAttribute(attribute, path.addLevel(attribute.name)))
+  def getTableAttributes(path: EntityPath)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[Attribute]]] =
+    getAttributesHelper(path)(tezosPlatformDiscoveryOperations.getTableAttributes)
+
+  // fetches table attributes without updating cache
+  def getTableAttributesWithoutUpdatingCache(path: EntityPath)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[Attribute]]] =
+    getAttributesHelper(path)(tezosPlatformDiscoveryOperations.getTableAttributesWithoutUpdatingCache)
+
+  // fetches attributes with given function
+  private def getAttributesHelper(path: EntityPath)(getAttributes: String => Future[Option[List[Attribute]]])
+    (implicit apiExecutionContext: ExecutionContext): Future[Option[List[Attribute]]] = {
+    for {
+      exists <- exists(path)
+      attributes <- getAttributes(path.entity)
+    } yield attributes
+      .filter { _ => exists }
```

Comment:
To be more exact you can move this `filter` to a `for` section as an `if` statement.

Reply:
are you sure? I think because this is a `Future` in for-comprehension and if this `if` statement fails every time we will get `Fututre.failed` instead of `Future.successful(None)`

Reply:
ok, changed